### PR TITLE
Clean up documentation

### PR DESCRIPTION
**Pull request by lukegb**
Performs some spring-cleaning of the documentation, including:
* Deleting outdated GSoC 2014 page
* Fixing some section header depth warnings
* Adding some more documentation to index.rst (and thereby the sidenav)
* Removing references to things which no longer exist

This removes two management commands (`import_intersphinx` and `sync_builds`) which no longer work and cannot be imported, because they rely on code which has already been deleted.

### CODE REVIEW

**Comment by lukegb** [comment]
These commits at least make `make html` warning-free (apart from the docstring from builds.models.Version), which is almost in Numpy docstring format (but is, afaict, the only class documented in such a way). I'll think about that one.

File: docs/index.rst [hunk 1]
```diff
@@ -7,7 +7,7 @@ We pull your code from your Subversion_, Bazaar_, Git_, and Mercurial_ repositor
 Then we build documentation and host it for you.
 Think of it as *Continuous Documentation*.
 
-The code is open source, and `available on github`_.
+The code is open source, and `available on GitHub`_.
```
Comment (ericholscher) [inline docs/index.rst]: This is a link, so the link text should be updated.

File: readthedocs/core/management/commands/import_intersphinx.py [hunk 2]
```diff
@@ -1,11 +0,0 @@
...
-
-from readthedocs.builds.constants import LATEST
-from readthedocs.builds.models import Version
-from readthedocs.projects.tasks import update_intersphinx
-
-
-class Command(BaseCommand):
-    def handle(self, *args, **options):
-        for version in Version.objects.filter(slug=LATEST):
-            update_intersphinx(version.pk)
```
Comment (ericholscher) [inline readthedocs/core/management/commands/import_intersphinx.py]: Any reason these files got deleted?

Comment (lukegb) [inline readthedocs/core/management/commands/import_intersphinx.py]: Per b9a7eb668cda98c9ca1d0d2c27f96e892b7608cb and 8adb9837295b861c3838338ba0b7b30aa9019156, these both rely on code that was deleted in 2014 (see the commit message, I'll add it to the PR description too), so they don't even import without throwing an error any more.

autodoc was trying to import them and throwing warnings.

File: docs/index.rst (hunk 1 above)
Comment (lukegb) [inline docs/index.rst]: Oops, good catch. Thanks.

**Comment by ericholscher** [comment]
Thanks for this!

**Pull request merged**